tests/components/proximity/test_init.py

```diff
@@ -1,557 +1,589 @@
-        )
-        self.hass.block_till_done()
-        self.hass.states.set(
-            "device_tracker.test2", "not_home", {"friendly_name": "test2"}
-        )
-        self.hass.block_till_done()
-
-        assert proximity.setup(
-            self.hass,
-            {
-                "proximity": {
-                    "home": {
-                        "ignored_zones": ["work"],
-                        "devices": ["device_tracker.test1", "device_tracker.test2"],
-                        "zone": "home",
-                    }
-                }
-            },
-        )
-
-        self.hass.states.set(
-            "device_tracker.test1",
-            "not_home",
-            {"friendly_name": "test1", "latitude": 20.1, "longitude": 10.1},
-        )
-        self.hass.block_till_done()
-        state = self.hass.states.get("proximity.home")
-        assert state.attributes.get("nearest") == "test1"
-        assert state.attributes.get("dir_of_travel") == "unknown"
-
-        self.hass.states.set(
-            "device_tracker.test2",
-            "not_home",
-            {"friendly_name": "test2", "latitude": 10.1, "longitude": 5.1},
-        )
-        self.hass.block_till_done()
-        state = self.hass.states.get("proximity.home")
-        assert state.attributes.get("nearest") == "test2"
-        assert state.attributes.get("dir_of_travel") == "unknown"
-
-        self.hass.states.set(
-            "device_tracker.test2",
-            "work",
-            {"friendly_name": "test2", "latitude": 12.6, "longitude": 7.6},
-        )
-        self.hass.block_till_done()
-        state = self.hass.states.get("proximity.home")
-        assert state.attributes.get("nearest") == "test1"
-        assert state.attributes.get("dir_of_travel") == "unknown"
+        },
+    )
+
+    hass.states.async_set(
+        "device_tracker.test1",
+        "not_home",
+        {"friendly_name": "test1", "latitude": 20.1, "longitude": 10.1},
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("proximity.home")
+    assert state.attributes.get("nearest") == "test1"
+    assert state.attributes.get("dir_of_travel") == "unknown"
+
+    hass.states.async_set(
+        "device_tracker.test2",
+        "not_home",
+        {"friendly_name": "test2", "latitude": 10.1, "longitude": 5.1},
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("proximity.home")
+    assert state.attributes.get("nearest") == "test2"
+    assert state.attributes.get("dir_of_travel") == "unknown"
+
+    hass.states.async_set(
+        "device_tracker.test2",
+        "work",
+        {"friendly_name": "test2", "latitude": 12.6, "longitude": 7.6},
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("proximity.home")
+    assert state.attributes.get("nearest") == "test1"
+    assert state.attributes.get("dir_of_travel") == "unknown"
+
+
+def config_zones(hass):
```

Comment:
This works. Alternative would have been to make it a fixture and use that in tests 👍 

```
@pytest.fixture
def hass_with_zones(hass):
    hass.config.components.add("zone")
    hass.states.async_set(…)
    hass.states.async_set(…)
    return hass
```